### ISSUE TITLE
Promise

### ISSUE THREAD

**Issue by ImTheDeveloper** [enricostara/termgram#1]
Hi!

Just wanted to raise this issue for the following error:

```
TELEGRAM.LINK
v.0.3.0

Welcome to TERMGRAM, a terminal client to connect with Telegram.

To protect your Telegram identity you must create a local user/password,
Termgram will use your password to encrypt your authorization data saved locally.
Choose your username, it will be local and it will not be used by Telegram!


/usr/local/lib/node_modules/termgram/lib/user-interface.js:17
    return new Promise(function (fulfill) {
               ^
ReferenceError: Promise is not defined
    at Object.askWordInput (/usr/local/lib/node_modules/termgram/lib/user-interface.js:17:16)
    at askUsername (/usr/local/lib/node_modules/termgram/lib/use-case/sign-up.js:51:8)
    at signUp (/usr/local/lib/node_modules/termgram/lib/use-case/sign-up.js:25:5)
    at main (/usr/local/lib/node_modules/termgram/termgram.js:31:9)
    at Object.<anonymous> (/usr/local/lib/node_modules/termgram/termgram.js:62:1)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
```

I appreciate that this is probably still alpha but wanted to ensure you knew there was an issue on ubuntu

**Comment by enricostara** [comment]
Which version of node have you installed?

Il sabato 2 maggio 2015, MrMobberley notifications@github.com ha scritto:

**Comment by ImTheDeveloper** [comment]
chris@chris-Inspiron-1545:~$ node -v
v0.10.25

**Comment by enricostara** [comment]
Promise are available by default starting from the version **0.12**

**telegram.link library needs only the 0.10**, but I'm writing this example application (and only for this) using also the promises for the sake of readability/semplicity
